src/HucaresServer.Storage.UnitTests/MissingLicensePlateTests.cs

```diff
@@ -0,0 +1,213 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using HucaresServer.Storage.Helpers;
+using FakeItEasy;
+using HucaresServer.Storage.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Shouldly;
+
+namespace HucaresServer.Storage.UnitTests
+{
+    [TestClass]
+    public class MissingLicensePlateTests
+    {
+        [TestMethod]
+        public void InsertMissingPlate_WhenPlateNumberIsValid_ShouldSucceedAndReturnExpected()
+        {
+            //Arrange
+            var fakeIQueryable = new List<MissingLicensePlate>().AsQueryable();
+            var fakeDbSet = A.Fake<DbSet<MissingLicensePlate>>();
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.MissingLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var missingLicensePlateHelper = new MissingPlateHelper(fakeDbContextFactory);
+
+            //Act
+            var expectedPlate = "JBA:555";
+            var dateInput = "Jan 1, 2018";
+            DateTime expectedDate = DateTime.Parse(dateInput);
+            var result = missingLicensePlateHelper.InsertPlateRecord(expectedPlate, expectedDate);
+
+            //Assert
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .MustHaveHappenedOnceExactly();
+
+            A.CallTo(() => fakeDbSet.Add(result))
+                .MustHaveHappened();
+
+            A.CallTo(() => fakeHucaresContext.SaveChanges())
+                .MustHaveHappenedOnceExactly();
+
+            result.PlateNumber.ShouldBe(expectedPlate);
+            result.SearchStartDateTime.ShouldBe(expectedDate);
+        }
+
+        [TestMethod]
+        public void InsertMissingPlate_WhenPlateNumberIsNotValid_ShouldThrowAnError()
+        {
+            //Arrange
+            var fakeDbSet = A.Fake<DbSet<MissingLicensePlate>>();
+
+            var fakeHucaresContext = A.Fake<HucaresContext>();
+            A.CallTo(() => fakeHucaresContext.MissingLicensePlates)
+                .Returns(fakeDbSet);
+
+            var fakeDbContextFactory = A.Fake<IDbContextFactory>();
+            A.CallTo(() => fakeDbContextFactory.BuildHucaresContext())
+                .Returns(fakeHucaresContext);
+
+            var missingPlateHelper = new MissingPlateHelper(fakeDbContextFactory);
+
+            var dateInput = "Jan 1, 2018";
+            DateTime expectedDate = DateTime.Parse(dateInput);
+            
+            //Act & Assert
+            Assert.ThrowsException<UriFormatException>(() => missingPlateHelper.InsertPlateRecord("5555555", expectedDate));
```

Comment:
InsertPlateRecord never throws UriFormatException, because it does not do any uri check. If you expect a wrong plate number to throw an exception, then implement a check for it.